pallets/connectors/src/inbound.rs

```diff
@@ -0,0 +1,270 @@
+
+		// Determine post adjustment amount
+		let pre_amount = T::ForeignInvestment::investment(&investor, invest_id.clone())?;
+		let post_amount = pre_amount.ensure_add(amount)?;
+
+		// Mint additional amount
+		T::Tokens::mint_into(currency, &investor, amount)?;
+
+		T::ForeignInvestment::update_investment(&investor, invest_id, post_amount)?;
+
+		Ok(())
+	}
+
+	/// Decreases an existing investment order of the investor.
+	///
+	/// Directly burns the decreased investment amount from the investor
+	/// account.
+	///
+	/// Initiates a return `ExecutedDecreaseInvestOrder` message to refund the
+	/// decreased amount on the source domain. The dispatch of this message is
+	/// delayed until the execution of the investment, e.g. at least until the
+	/// next epoch transition.
+	pub fn do_decrease_invest_order(
+		pool_id: PoolIdOf<T>,
+		tranche_id: TrancheIdOf<T>,
+		investor: T::AccountId,
+		currency_index: GeneralCurrencyIndexOf<T>,
+		amount: <T as Config>::Balance,
+	) -> DispatchResult {
+		// Retrieve investment details
+		let invest_id: T::TrancheCurrency = Self::derive_invest_id(pool_id, tranche_id)?;
+		let currency = Self::try_get_payment_currency(invest_id.clone(), currency_index)?;
+
+		// Determine post adjustment amount
+		let pre_amount = T::ForeignInvestment::investment(&investor, invest_id.clone())?;
+		let post_amount = pre_amount.ensure_sub(amount)?;
+
+		T::ForeignInvestment::update_investment(&investor, invest_id, post_amount)?;
+
+		// Burn decreased amount
+		T::Tokens::burn_from(currency, &investor, amount)?;
+
+		// TODO(subsequent PR): Handle response `ExecutedDecreaseInvestOrder`message to
+		// source destination which should refund the decreased amount.
+		// Blocked by https://github.com/centrifuge/centrifuge-chain/pull/1363
+		// Should be handled by pallet-foreign-investments
+
+		Ok(())
+	}
+
+	/// Increases an existing redemption order of the investor.
+	///
+	/// Transfers the decreased redemption amount from the holdings of the
+	/// `DomainLocator` account of origination domain of this message into the
+	/// investor account.
+	///
+	/// Assumes that the amount of tranche tokens has been locked in the
+	/// `DomainLocator` account of the origination domain beforehand.
+	pub fn do_increase_redemption(
+		pool_id: PoolIdOf<T>,
+		tranche_id: TrancheIdOf<T>,
+		investor: T::AccountId,
+		amount: <T as Config>::Balance,
+		sending_domain: DomainAddress,
+	) -> DispatchResult {
+		// Retrieve investment details
+		let invest_id: T::TrancheCurrency = Self::derive_invest_id(pool_id, tranche_id)?;
+
+		// Determine post adjustment amount
+		let pre_amount = T::ForeignInvestment::redemption(&investor, invest_id.clone())?;
+		let post_amount = pre_amount.ensure_add(amount)?;
+
+		// Transfer tranche tokens from `DomainLocator` account of origination domain
+		T::Tokens::transfer(
+			invest_id.clone().into(),
+			&T::AccountConverter::convert(sending_domain),
```

Comment:
Again, I think this should be `DomainLocator`, right?

Reply:
[cbba52c](https://github.com/centrifuge/centrifuge-chain/pull/1364/commits/cbba52ca61042bbf52ca841a4f8338d0d4e0630f)